### PR TITLE
[Transform] Log and audit a warning if all the group-by fields are runtime fields

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR adds logging and auditing for the case when all the fields that transform uses as group-by (in `pivot`) or unique key (in `latest`) are search runtime fields.

This PR is marked as `>non-issue` as it enhances an unreleased feature.

Relates https://github.com/elastic/elasticsearch/issues/65147

### CODE REVIEW

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/Pivot.java [hunk 1]
```diff
@@ -77,6 +79,11 @@ public void validateConfig(ActionListener<Boolean> listener) {
         listener.onResponse(true);
     }
 
+    @Override
+    public List<String> getPerformanceCriticalFields() {
+        return config.getGroupConfig().getGroups().values().stream().map(SingleGroupSource::getField).collect(toList());
```
Comment (benwtrent) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/Pivot.java:84]: I am not sure all of these would be performance critical. it almost seems like they would be performance critical IF they didn't support missing buckets right?

If the group config supported missing buckets, we don't get the benefit of filtering for changes or something? I remember there was a trade off. 

Right @hendrikmuhs

Comment (hendrikmuhs) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/Pivot.java:84]: > If the group config supported missing buckets, we don't get the benefit of filtering for changes or something? I remember there was a trade off.

missing bucket runs a bit slower in composite aggs, but I think we can ignore that, its like 10-20% improvement. Problematic are group_by with scripts, but we already have a warning for that: We warn if change detection is completely disabled, which means we do a full re-run on every checkpoint.

**Comment by hendrikmuhs** [comment]
Can you also add a check if the timestamp field configured for `sync` is a search runtime field?

**Comment by przemekwitek** [comment]
> Can you also add a check if the timestamp field configured for `sync` is a search runtime field?

Done.

**Review by hendrikmuhs — APPROVED** [review]
LGTM

**Pull request merged**